### PR TITLE
Use a published GraphQL parser

### PR DESCRIPTION
**Pull request by serras**
Finally we can drop the dependency on a particular commit of a repo!

### CODE REVIEW

File: graphql/src/Mu/GraphQL/Annotations.hs [hunk 1]
```diff
@@ -110,11 +108,14 @@ instance (ReflectValueConst x, ReflectValueConstList xs)
...
   reflectValueConstObject _ = []
 instance (KnownSymbol a, ReflectValueConst x, ReflectValueConstObject xs)
          => ReflectValueConstObject ( '(a, x) ': xs) where
   reflectValueConstObject _
-    = GQL.ObjectFieldG (coerce $ T.pack $ symbolVal (Proxy @a)) (reflectValueConst (Proxy @x))
+    = GQL.ObjectField (T.pack $ symbolVal (Proxy @a))
+                      (GQL.Node (reflectValueConst (Proxy @x)) zl)
+                      zl
       : reflectValueConstObject (Proxy @xs)
+    where zl = GQL.Location 0 0
```
Comment (kutyel) [inline graphql/src/Mu/GraphQL/Annotations.hs:121]: I don't understand this bit 🤔

File: graphql/src/Mu/GraphQL/Server.hs [hunk 2]
```diff
@@ -175,6 +175,7 @@ httpGraphQLAppTrans f server q m s req res =
     toResponse = res . responseBuilder ok200 [] . T.encodeUtf8Builder . encodeToLazyText
     unpackUnicodeException :: UnicodeException -> T.Text
     unpackUnicodeException (DecodeError str _) = T.pack str
+    unpackUnicodeException _                   = error "EncodeError is deprecated"
```
Comment (kutyel) [inline graphql/src/Mu/GraphQL/Server.hs:178]: 👍🏻

File: graphql/src/Mu/GraphQL/Server.hs [hunk 3]
```diff
@@ -57,6 +56,7 @@ import           Network.Wai.Handler.Warp         (Port, Settings, run, runSetti
 import qualified Network.Wai.Handler.WebSockets   as WS
 import qualified Network.WebSockets               as WS
 
+import           Mu.GraphQL.Quasi.LostParser      (parseDoc)
```
Comment (kutyel) [inline graphql/src/Mu/GraphQL/Server.hs:59]: Is my "lost parser" still needed with this new parser? 🤔

**Review by kutyel — APPROVED** [review]
So much less `coerce`! 🥳

File: graphql/src/Mu/GraphQL/Annotations.hs (hunk 1 above)
Comment (serras) [reply graphql/src/Mu/GraphQL/Annotations.hs:121]: This library requires a location for each node, so I'm creating a fake one. I'm planning to take advantage of these locations in error messages in the future, but now I just wanted to get this out.

File: graphql/src/Mu/GraphQL/Server.hs (hunk 3 above)
Comment (serras) [reply graphql/src/Mu/GraphQL/Server.hs:59]: Now it's more like "function which runs the parser and juggles a bit with the result so it pleases what we need", but I decided to keep the name xD

**Pull request merged**